nautobot/docs/user-guide/platform-functionality/gitrepository.md

```diff
@@ -351,3 +351,137 @@ Export templates may be provided as files located in `/export_templates/<groupin
+
+### List Existing Repositories
+
+Just like other Nautobot apps and models, it is possible to use the Nautobot REST API to list existing configured repositories by issuing a `GET` request to the model's *list* endpoint. As usual, objects are listed under the response object's `results` parameter:
+
+```no-highlight
+curl -s -X GET \
+-H "Authorization: Token $TOKEN" \
+-H "Content-Type: application/json" \
+-H "Accept: application/json; version=2.4" \
+http://nautobot/api/extras/git-repositories/ | jq '.'
+```
+
+### Describe a Specific Existing Repository
+
+If you have the `id` of a specific record from a previous `GET` operation or elsewhere, you can return the specific details for the record with a `GET` request referencing the specific `id`:
+
+```no-highlight
+curl -s -X GET \
+-H "Authorization: Token $TOKEN" \
+-H "Content-Type: application/json" \
+-H "Accept: application/json; version=2.4" \
+http://nautobot/api/extras/git-repositories/2ecb8556-db58-466d-8278-860b8fd74627/ | jq '.'
+```
+
+Returns, for example:
+
+```json
+{
+  "id": "2ecb8556-db58-466d-8278-860b8fd74627",
+  "object_type": "extras.gitrepository",
+  "display": "demo-git-datasource",
+  "url": "http://nautobot/api/extras/git-repositories/2ecb8556-db58-466d-8278-860b8fd74627/",
+  "natural_slug": "demo-git-datasource_2ecb",
+  "provided_contents": [
+    "extras.configcontext",
+    "extras.job",
+    "extras.exporttemplate",
+    "extras.configcontextschema"
+  ],
+  "name": "demo-git-datasource",
+  "slug": "demo_git_datasource",
+  "remote_url": "https://github.com/nautobot/demo-git-datasource.git",
+  "branch": "main",
+  "current_head": "939ea1ed854e405b600d70f798804eb2da356231",
+  "secrets_group": null,
+  "created": "2025-01-21T12:17:24.945117Z",
+  "last_updated": "2025-01-21T12:17:30.081417Z",
+  "notes_url": "http://nautobot/api/extras/git-repositories/2ecb8556-db58-466d-8278-860b8fd74627/notes/",
+  "custom_fields": {},
+  "tags": []
+}
+```
+
+!!! note
+    By design and like other secrets in Nautobot, the tokens used for a Git repository are never retrievable via a REST API request.
+
+### Trigger a Sync or Resync of a Defined Repository
+
+As noted previously, a newly defined Git repository will not automatically sync immediately, so if you want it to sync or resync you can issue a `POST` request to the specific `id` followed by `/sync/` like this:
+
+```no-highlight
+curl -s -X POST \
+-H "Authorization: Token $TOKEN" \
+-H "Content-Type: application/json" \
+-H "Accept: application/json; version=2.4" \
+http://nautobot/api/extras/git-repositories/2ecb8556-db58-466d-8278-860b8fd74627/sync/
+```
+
+Which returns, for example:
+
+```no-highlight
+{
+  "message": "Repository demo-git-datasource sync job added to queue."
```

Comment:
Wow, is that seriously all we reply with? Not even the job-result ID to query for status? That's definitely something we should enhance...

Reply:
Haha sadly it is! I've created #6808 and assigned it to myself to improve this when I have free time.